### DISCUSSION TITLE
Specifying a custom file name for mg_http_upload()

### DISCUSSION THREAD

**Discussion by dmitry-kandiner** [cesanta/mongoose#2831] [Ask questions, get answers]
Hello!

In version 7.12, the `mg_http_upload()` function accepted a path as its parameter. It seems that starting from version 7.13 the function accepts a directory name instead, and taking the file name from the "**name**" variable of the GET query.
Is there a way to specify a temporary file name instead (other than the name provided in the provided in the query)?

Best regards,
Dmitry

**Comment by scaprile** [comment]
There is no such function.
If you mean `mg_http_upload()`, this is its doc https://mongoose.ws/documentation/#mg_http_upload and this tutorial shows how to use it:
https://mongoose.ws/documentation/tutorials/http/file-uploads/#binary-upload-multiple-posts, along with this example https://github.com/cesanta/mongoose/tree/master/tutorials/http/file-upload-multiple-posts
Should there be other way, it would be explained in our docs.
Please see our [documentation](https://mongoose.ws/documentation/),  and follow the guidelines in our [tutorials](https://mongoose.ws/documentation/#tutorials-and-examples).

https://github.com/cesanta/mongoose/blob/22b52f831bfd103b70a9b27209a0d2dd7df2fe1e/src/http.c#L932-L974

**Reply by dmitry-kandiner** [reply]
@scaprile , 
Of course, I meant the `mg_http_upload()` function. Still, in an older version of it (see the [diff](https://github.com/cesanta/mongoose/commit/e0668bc1c01804859b8ce98d902a23e32efe5c10#diff-4b8a915e7a0661ce6c35078f460d190a3d7212e7bf57113a26e25b74f3d4e41bL917) ) it accepted a file name with the path as its parameter, while now it only accepts the path, and the file name is extracted from the query itself.
This blocks the option for specifying a custom file name, say, for a temporary file, instead of the one provided in the query.
Be sure I read through the tutorials before posting here.
Best regards, 
Dmitry
